### ISSUE TITLE
ERROR: Project name expected 1 dots, supplied 'foo.bar.content' has 2 dots

### ISSUE THREAD

**Issue by regebro** [collective/templer.plone#9]
By restricting the choice in namespace depth you make the template useless for anyone that doesn't want to use the exact same amount of nsamespaces as the template author.

**Comment by claytron** [comment]
iirc @MatthewWilkes had something that could dynamically do namespacing in his custom templates.

+1 for adding support to do that.

**Comment by cewing** [comment]
Hi Lennart.  Clayton

support for the templer script, one- and two-namespace packages and anything else you'd need to hack this in is in templer.core.  I'd welcome the addition.  However, we did not pursue this in ZopeSkel because the types of folks who want/need a disconnect between the namespaces they provide in a name and the ones that exist on the filesystem are the types of expert programmers who don't really need a package template to get going.  

That being said, if anyone has clever, sustainable ideas on how to do this and wants to support it, templer.core is the right place to do so.

**Comment by regebro** [comment]
Now I'm confused. There is no disconnect between anything in this. (Sorry for the close/reopen, I pushed the wrong button).

**Comment by cewing** [comment]
Ahh, I see I misunderstood your issue.  You mean that an author would simply provide a name and get a package of the appropriate namespace depth?  So If I said `templer plone my.super.nested.package` I'd get a package with three namespaces and the package?  

If @MatthewWilkes has in fact solved this, I'd be glad to see it included.  

It would belong in templer.core, as that is where the basic package structure templates are set out.  It'd be really nice not to have to duplicate things like a GS profile structure too.  

Anyone have time to hack this?  Lennart?

**Comment by regebro** [comment]
Yes, that's what I meant. I would like not to be forced to a particular package namespace depth. At least in ZopeSkel there was the additional problem that this depth was different for different templates, so that the portlet template required two dots, and the archetype one only one, which makes it really hard to use both in the same project.

No, sorry, I don't see that I have time to hack that this side of the heat death of the universe. :-)

**Comment by cewing** [comment]
I'd like to think in templer it's at least a bit better since the template names give you a hint.  Also, with inline validation of your answers you should be getting this feedback immediately upon trying to run the command.  You are using `bin/templer` or `bin/zopeskel` right?  Not `bin/paster`?

**Comment by regebro** [comment]
I do run bin/templer, yes. Otherwise I think we are talking around each other again. I don't see any hints, nor do I see how hints would help. I do get feedback immediately upon running the command as per the topic:

"ERROR: Project name expected 1 dots, supplied 'foo.bar.content' has 2 dots"

That's perfectly clear. What would help is for it to work. :-)
